src/vs/editor/contrib/hover/hover.ts

```diff
@@ -258,14 +270,198 @@ class ShowHoverAction extends EditorAction {
 	}
 
 	public run(accessor: ServicesAccessor, editor: ICodeEditor): void {
+		const textModelResolverService = accessor.get(ITextModelService);
+		const modeService = accessor.get(IModeService);
+
 		let controller = ModesHoverController.get(editor);
 		if (!controller) {
 			return;
 		}
 		const position = editor.getPosition();
+		const word = editor.getModel().getWordAtPosition(position);
 		const range = new Range(position.lineNumber, position.column, position.lineNumber, position.column);
```

Comment:
we can move the `range` creation downwards to the line before `controller.showContentHover`.